src/ReactHardwareComponent.js

```diff
@@ -92,45 +115,40 @@ ReactHardwareComponent.Mixin = {
     const prevElement = this._currentElement;
     this._currentElement = nextElement;
 
-    if (this._currentElement.props[CONTAINER_KEY] === CONTAINER_VALUE) {
-      this.updateChildren(nextElement.props.children, transaction, context);
-      return;
-    }
-
-    const updatePayload = Object.assign(
-      {},
-      nextElement.props,
-      diff(
-        prevElement.props,
-        nextElement.props,
-        this.viewConfig.validAttributes
-      )
-    );
-
-    if (process.env.NODE_ENV !== 'production') {
-      if (prevElement.props.pin) {
-        invariant(
-          prevElement.props.pin === nextElement.props.pin,
-          'A mounted component cannot be mounted into a new Pin. The `pin` ' +
-          'attribute is immutable. Check the render function of ' +
-          nextElement.displayName  + '.' // TODO
+    // special cases
+    switch (nextElement.type) {
```

Comment:
for now since we only have one `case` I’d like to revert this to a simple `if (nextElement.type === 'pin')`